servicetalk-utils-internal/src/main/java/io/servicetalk/utils/internal/DurationUtils.java

```diff
@@ -54,4 +65,17 @@ public static Duration ensurePositive(final Duration duration, final String name
         }
         return duration;
     }
+
+    /**
+     * Converts a {@code Duration} to nanoseconds or if the resulting value would overflow a 64-bit signed integer then
+     * either {@code Long.MIN_VALUE} or {@code Long.MAX_VALUE} as appropriate.
+     *
+     * @param duration the duration to convert
+     * @return the converted nanoseconds value
+     */
+    public static long toNanos(final Duration duration) {
```

Comment:
does it make sense to qualify the name to indicate under/overflow protection? For example in `FlowControlUtils` we use [WithUnderOverflowProtection](https://github.com/apple/servicetalk/blob/main/servicetalk-concurrent-internal/src/main/java/io/servicetalk/concurrent/internal/FlowControlUtils.java#L122).

Reply:
+1, let's keep as-is. It's an internal method anyway